core/include/drivers/rstctrl.h

```diff
@@ -0,0 +1,228 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, Linaro Limited
+ *
+ * Self contained reset controller framework in rstctrl.h
+ * A reset controller exposes reset level controls to a
+ * platform.
+ */
+
+#ifndef __DRIVERS_RSTCTRL_H
+#define __DRIVERS_RSTCTRL_H
+
+#include <kernel/dt.h>
+#include <kernel/dt_driver.h>
+#include <stdint.h>
+#include <tee_api_types.h>
+
+struct rstctrl;
+
+struct rstctrl_ops {
+	/*
+	 * Operators on reset control(s) exposed by a reset controller
+	 *
+	 * assert_level: Assert reset level on control with a timeout hint
+	 * deassert_level: Deassert reset level on control with a timeout hint
+	 * get_name: Get a string name for the controller, or NULL is none
+	 *
+	 * @rstctrl: Reset control reference
+	 * @id: Identifier for the reset level control in the reset controller
+	 * @to_ms: Timeout in microsencond or RSTCTRL_NO_TIMEOUT, may be ignored
```

Comment:
> id and to_ms are not parameters

Right, I described them here since common arguments to the operators.
You would prefer another way to describe them?

> BTW, I do not see where you can specify the reset line to use. 

Consumer calls `rstctrl_dt_get_by_index()` (or `rstctrl_dt_get_by_name()`, as clocks) and gets a reset line, I.e. a `struct rstctrl` pointer.
It then calls `rstctrl_assert()` to request request.
It may (often) need to call `rstctrl_deassert()` when the line shall be manually released.

Should rstctrl_dt_get_by_index() return a rstctrl reference + a line ID reference, to be used in rstctrl_*() API functions?